### PR TITLE
Add Cinny

### PR DESCRIPTION
**Pull request by kfiven**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by kfiven** [comment]
bot, build in.cinny.Cinny

File: in.cinny.Cinny.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+command: cinny
+rename-icon: cinny
+
+modules:
+  - shared-modules/libappindicator/libappindicator-gtk3-12.10.json
+
+  - name: Cinny
+    buildsystem: simple
+    build-commands:
+      - ls -a
```
Comment (hfiguiere) [inline in.cinny.Cinny.yml:14]:
```suggestion
```

File: in.cinny.Cinny.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      - mkdir -p /app/share/icons/hicolor
+      - cp -r usr/share/icons/hicolor/* /app/share/icons/hicolor
+      - rm -rf /app/share/icons/hicolor/512x512@2x
+      - install -Dm744 usr/share/applications/cinny.desktop /app/share/applications/in.cinny.Cinny.desktop
+    sources:
+      - url: https://github.com/cinnyapp/cinny-desktop/releases/download/v2.0.2/cinny_2.0.2_amd64.deb
+        sha256: d933cdf55640c441a4538900e52c9d1804c8639e8568c81b464a1b4896bd3741
+        type: file
+    cleanup:
+      - usr
```
Comment (hfiguiere) [inline in.cinny.Cinny.yml:29]: this doesn't do what you think it does.

File: in.cinny.Cinny.desktop [hunk 3]
```diff
@@ -0,0 +1,10 @@
+[Desktop Entry]
```
Comment (hfiguiere) [inline in.cinny.Cinny.desktop:1]: you don't need this file you should remove it.

File: in.cinny.Cinny.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+finish-args:
```
Comment (hfiguiere) [inline in.cinny.Cinny.yml:31]: finish-args is usually above modules.

File: in.cinny.Cinny.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  # Required due to being a GUI application
+  - --socket=x11
```
Comment (hfiguiere) [inline in.cinny.Cinny.yml:33]: if this app supports wayland this should be:
```suggestion
  - --socket=fallback-x11
```

File: in.cinny.Cinny.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  # Required to make sure x11 performance is achived on all platforms
+  # At least that's what the legends tell. it might be worth experimenting
+  # with dropping this permission.
+  - --share=ipc
+  # Required for experimental wayland support
+  - --socket=wayland
```
Comment (hfiguiere) [inline in.cinny.Cinny.yml:39]: by experimental, you it is not working by default then you shouldn't allow it.

File: in.cinny.Cinny.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,81 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2022 Ajay Bura <ajbura@gmail.com> -->
+<component type="desktop">
+    <id>in.cinny.Cinny.desktop</id>
```
Comment (hfiguiere) [inline in.cinny.Cinny.appdata.xml:4]:
```suggestion
    <id>in.cinny.Cinny</id>
```

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: in.cinny.Cinny.yml [hunk 8]
```diff
@@ -0,0 +1,60 @@
+id: in.cinny.Cinny
+runtime: org.gnome.Platform
+runtime-version: '42'
+sdk: org.gnome.Sdk
+command: cinny
+rename-icon: cinny
+
+modules:
+  - shared-modules/libappindicator/libappindicator-gtk3-12.10.json
```
Comment (hfiguiere) [inline in.cinny.Cinny.yml:9]: you haven't added the `shared-modules`.

**Comment by kfiven** [comment]
bot, build in.cinny.Cinny

File: in.cinny.Cinny.yml [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+      - rm -f cinny_*.deb
+      - tar xf data.tar.gz
+      - rm -f control.tar.gz data.tar.xz debian-binary
+      - cp -r usr/bin/cinny /app/bin
+      - mkdir -p /app/share/icons/hicolor
+      - cp -r usr/share/icons/hicolor/* /app/share/icons/hicolor
+      - rm -rf /app/share/icons/hicolor/512x512@2x
+      - install -Dm744 usr/share/applications/cinny.desktop /app/share/applications/in.cinny.Cinny.desktop
+    sources:
+      - url: https://github.com/cinnyapp/cinny-desktop/releases/download/v2.0.2/cinny_2.0.2_amd64.deb
```
[the remark this answers is not in the document]
Comment (kfiven) [reply in.cinny.Cinny.yml:33]: Fixed.

I have tried building from sources but had issues [generating sources](https://github.com/flatpak/flatpak-builder-tools/issues/278).

File: flathub.json [hunk 10]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: this is disappointing, really.

File: in.cinny.Cinny.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply in.cinny.Cinny.yml:33]: maybe upstream can build it then?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: flathub.json (hunk 10 above)
Comment (kfiven) [reply flathub.json:2]: isn't it how it is supposed to be if we are only building for x86_64?

File: in.cinny.Cinny.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (kfiven) [reply in.cinny.Cinny.yml:33]: is there some doc that can help with that?

File: flathub.json (hunk 10 above)
Comment (hfiguiere) [reply flathub.json:2]: it remain it is disappointing.

File: in.cinny.Cinny.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply in.cinny.Cinny.yml:33]: you are asking me how to build that project?

[the remark this answers is not in the document]
Comment (kfiven) [reply in.cinny.Cinny.yml:33]: I know that but I am asking how to build flatpak sources on the upstream side?

File: flathub.json (hunk 10 above)
Comment (kfiven) [reply flathub.json:2]: ah now I get what you meant by that, yeah it is, so I am waiting for Tauri action to support producing aarch64 binaries.

File: in.cinny.Cinny.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply in.cinny.Cinny.yml:33]: how would I know? Like it is build for x86_64?

[the remark this answers is not in the document]
Comment (kfiven) [reply in.cinny.Cinny.yml:33]: yeah it build for x86_64.

File: in.cinny.Cinny.yml [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
```
Comment (barthalion) [inline in.cinny.Cinny.yml:33]: Please configure https://github.com/flathub/flatpak-external-data-checker/ to automate updates.

File: in.cinny.Cinny.yml [hunk 12]
```diff
@@ -0,0 +1,37 @@
+id: in.cinny.Cinny
+runtime: org.gnome.Platform
```
Comment (barthalion) [inline in.cinny.Cinny.yml:2]: Does it actually need GNOME runtime or freedesktop would suffice here?

Comment (kfiven) [reply in.cinny.Cinny.yml:2]: It builds with freedesktop too. Are there benefits of using that over gnome one?

Comment (hfiguiere) [reply in.cinny.Cinny.yml:2]: smaller runtime is always better.

File: in.cinny.Cinny.yml (hunk 11 above)
Comment (kfiven) [reply in.cinny.Cinny.yml]: Application has auto-updates in itself, do I still need to configure this?

Comment (hfiguiere) [reply in.cinny.Cinny.yml]: you probaly need to disable them. The content of the flatpak IS read-only.

File: in.cinny.Cinny.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply in.cinny.Cinny.yml]: which is why I always advocate for building from source in the flatpak.

File: in.cinny.Cinny.yml (hunk 11 above)
Comment (kfiven) [reply in.cinny.Cinny.yml]: so if I get it correctly, that's only possible if I build flatpak from the source? (or it can be done for .deb too?)

**Comment by kfiven** [comment]
bot, build in.cinny.Cinny

**Comment by barthalion** [comment]
You need a metainfo file. You can use either https://bilelmoussaoui.github.io/metainfo-generator/ or https://www.freedesktop.org/software/appstream/metainfocreator/ and validate the file locally with `org.freedesktop.appstream-glib` available on Flathub. It should be installed at `/app/share/metainfo/$FLATPAK_ID.metainfo.xml`.

**Comment by kfiven** [comment]
bot, build in.cinny.Cinny

**Comment by kfiven** [comment]
bot, build in.cinny.Cinny

**Comment by kfiven** [comment]
bot, build in.cinny.Cinny

**Pull request closed without merging**